libqtile/command/base.py

```diff
@@ -40,6 +42,50 @@
     ItemT = Optional[tuple[bool, list[str | int]]]
 
 
+def expose_command(name: Callable | str | list[str] | None = None) -> Callable:
+    """
+    Decorator to expose commands to IPC.
+
+    The exposed command will have the name of the defined method.
+
+    Methods can also be exposed via multiple names by passing the names of
+    the additional method.
+
+    e.g. if a layout wants "up" and "previous" to call the
+    same method:
+
+    @expose_command("previous")
+    def up(self):
+        ...
+
+    `up` will be exposed as `up` and `previous`.
+
+    Multiple names can be passed as a list.
+    """
+
+    def wrapper(func: Callable):
+        setattr(func, "_cmd", True)
+        if name is not None:
+            if not hasattr(func, "_mapping"):
+                setattr(func, "_mapping", list())
+            if isinstance(name, list):
+                func._mapping += name  # type:ignore
+            elif isinstance(name, str):
+                func._mapping.append(name)  # type:ignore
+            else:
+                logger.error("Unexpected value received in command decorator: %s", name)
+        return func
+
+    # If the decorator is added with no parentheses then we should treat it
+    # as if it had been i.e. expose the decorated method
+    if callable(name):
+        func = name
+        name = None
+        return wrapper(func)
```

Comment:
Why not just `return wrapper(name)`?

Reply:
if `name` is not None then what happens to `wrapper` (lines 66-77). I had assumed I needed to clear `name` so as not to trigger an error in `wrapper`.

Reply:
Oh, right you are, I had incorrectly thought `wrapper` was using a different `name`. Ignore :)

Reply:
Sorry for getting hung up on this, I found it a bit difficult to reason with and I think I figured out why :smile: 

Where I've seen this logic in other places (e.g. the Python Cookbook), they do something like this:
```py
    if callable(name):
        return expose_command()(name)
```
This does create a fresh scope of locals so no need to reset `func` and `name`, which I think is a bit clearer to think about. If that's at the top of the method then the internal function doesn't get defined unnecessarily and means when reading the method you can follow the type of `name` and it's clear when you get to `def wrapper` that it isn't a callable.

Happy to go with what you've put already if you prefer that, I'm just putting this out there just in case :)